jquery.pjax.js

```diff
@@ -562,6 +562,21 @@ function uniqueId() {
 }
 
 function cloneContents(container) {
+  // Preserve textarea values
+  var textarea = findAll(container, "textarea")
+  textarea.text(function(i, text){return textarea[i].value})
+
+  // Preserve select values
+  findAll(container, "select").each(function(i, elem){
+    elem = $(elem);
+    var values = $(elem).val();
+    values = $.isArray(values) ? values : [values];
+    elem.find('option[selected]').attr('selected', false);
```

Comment:
I'm not happy that this performs DOM changes on elements before they are cloned. How about just making these changes on the elements after they have been cloned?